SonarQube.Client/Models/ServerSentEvents/ClientContract/ITaintServerEvent.cs

```diff
@@ -34,20 +34,26 @@ public interface ITaintServerEvent : IServerEvent
     public interface ITaintVulnerabilityRaisedServerEvent : ITaintServerEvent
     {
         string Branch { get; }
+        ITaintIssue Issue { get; }
     }
 
-    public class TaintVulnerabilityRaisedServerEvent : ITaintVulnerabilityRaisedServerEvent
+    internal class TaintVulnerabilityRaisedServerEvent : ITaintVulnerabilityRaisedServerEvent
     {
-        public TaintVulnerabilityRaisedServerEvent(string projectKey, string key, string branch)
+        public TaintVulnerabilityRaisedServerEvent(string projectKey, string key, string branch, 
+            //DateTimeOffset creationDate, 
+            string ruleKey, SonarQubeIssueSeverity severity, SonarQubeIssueType type,
```

Comment:
The original json has the fields in a flat structure, but there is also a separate new taint API (that we should use) and it returns the new structure.
I'm not sure if it's better to have the TaintVulnerabilityRaisedServerEvent be identical to the json (flat, and inheriting from ITaintIssue), or the way I did it (which requires a bit of a "hack" to convert the flat fields into the issue).

LMK what you think.

Reply:
Need more context, don't know what is best here

Reply:
Let's see what @duncanp-sonar says

Reply:
As discussed offline, it should make too much difference since we can change it later if necessary.
The consensus was that the nested version would be nicer to work with in SLVS so we'd start with that.